pkg/kubectl/describe.go

```diff
@@ -528,7 +529,30 @@ func describeContainers(pod *api.Pod, out io.Writer) {
 
 		fmt.Fprintf(out, "    Ready:\t%v\n", printBool(status.Ready))
 		fmt.Fprintf(out, "    Restart Count:\t%d\n", status.RestartCount)
+		fmt.Fprintf(out, "    Environment Variables:\t\n")
+		for _, e := range container.Env {
+			if e.ValueFrom != nil && e.ValueFrom.FieldRef != nil {
+				valueFrom := envValueFrom(pod, e)
+				fmt.Fprintf(out, "        \t%s\t = %s (%s:%s)\n", e.Name, valueFrom, e.ValueFrom.FieldRef.APIVersion, e.ValueFrom.FieldRef.FieldPath)
+			} else {
+				fmt.Fprintf(out, "        \t%s\t = %s\n", e.Name, e.Value)
+			}
+		}
+	}
+}
+
+func envValueFrom(pod *api.Pod, e api.EnvVar) string {
+	internalFieldPath, _, err := api.Scheme.ConvertFieldLabel(e.ValueFrom.FieldRef.APIVersion, "Pod", e.ValueFrom.FieldRef.FieldPath, "")
+	if err != nil {
+		return err.Error()
 	}
+
+	valueFrom, err2 := fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
```

Comment:
Was misunderstanding on my part. I wasn't familiar with FieldPath and didn't realize api validation would restrict it to supported paths on object creation. The comments you added should be helpful for the next reader.